File: abilities-common/src/main/java/io/github/solusmods/eternalcore/abilities/impl/network/EternalCoreAbilityNetwork.java
package io.github.solusmods.eternalcore.abilities.impl.network;

import io.github.solusmods.eternalcore.abilities.impl.network.c2s.*;
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;

public class EternalCoreAbilityNetwork {
    public static void init() {
        NetworkUtils.registerC2SPayload(RequestAbilityActivationPacket.TYPE, RequestAbilityActivationPacket.STREAM_CODEC, RequestAbilityActivationPacket::handle);
        NetworkUtils.registerC2SPayload(RequestAbilityReleasePacket.TYPE, RequestAbilityReleasePacket.STREAM_CODEC, RequestAbilityReleasePacket::handle);
        NetworkUtils.registerC2SPayload(RequestAbilityScrollPacket.TYPE, RequestAbilityScrollPacket.STREAM_CODEC, RequestAbilityScrollPacket::handle);
        NetworkUtils.registerC2SPayload(RequestAbilityTogglePacket.TYPE, RequestAbilityTogglePacket.STREAM_CODEC, RequestAbilityTogglePacket::handle);
    }
}

File: abilities-common/src/main/kotlin/io/github/solusmods/eternalcore/abilities/impl/network/EternalCoreAbilityNetwork.kt
package io.github.solusmods.eternalcore.abilities.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.abilities.impl.network.c2s.RequestAbilityActivationPacket
import io.github.solusmods.eternalcore.abilities.impl.network.c2s.RequestAbilityReleasePacket
import io.github.solusmods.eternalcore.abilities.impl.network.c2s.RequestAbilityScrollPacket
import io.github.solusmods.eternalcore.abilities.impl.network.c2s.RequestAbilityTogglePacket
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils

object EternalCoreAbilityNetwork {
    fun init() {
        NetworkUtils.registerC2SPayload<RequestAbilityActivationPacket?>(
            RequestAbilityActivationPacket.Companion.TYPE,
            RequestAbilityActivationPacket.Companion.STREAM_CODEC
        ) { obj: RequestAbilityActivationPacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        NetworkUtils.registerC2SPayload<RequestAbilityReleasePacket?>(
            RequestAbilityReleasePacket.Companion.TYPE,
            RequestAbilityReleasePacket.Companion.STREAM_CODEC
        ) { obj: RequestAbilityReleasePacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        NetworkUtils.registerC2SPayload<RequestAbilityScrollPacket?>(
            RequestAbilityScrollPacket.Companion.TYPE,
            RequestAbilityScrollPacket.Companion.STREAM_CODEC
        ) { obj: RequestAbilityScrollPacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        NetworkUtils.registerC2SPayload<RequestAbilityTogglePacket?>(
            RequestAbilityTogglePacket.Companion.TYPE,
            RequestAbilityTogglePacket.Companion.STREAM_CODEC
        ) { obj: RequestAbilityTogglePacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
    }
}
